plugins/module_utils/mso.py

```diff
@@ -1032,15 +1036,23 @@ def lookup_users(self, users, ignore_not_found_error=False):
 
     def get_user_from_list_of_users(self, user_name, list_of_users, login_domain=""):
```

Comment:
should we change the description of the function?
should we change the function name itself, since now it does not match the behaviour of dict or list?

my preference would be to write a new function for the new functionality which gets called with the type is instance check done before the call of the function?

I think we should change the argument name of list of users to just users.

Reply:
This function retrieves the user object directly from the API response, rather than taking a user's list as input, because of that, I did not modify the function name. Additionally, I have incorporated the new logic within the function and included notes for the new change.